### DISCUSSION TITLE
How do I have multiple trigger for `lua` snippet?

### DISCUSSION THREAD

**Discussion by budimanjojo** [L3MON4D3/LuaSnip#775] [General]
Hello! In `vscode` snippets, we can have multiple "prefix" to trigger the snippet. Is there a way to achieve this with `lua` snippet? `context.trig` is type of `string` so I don't think it's supported, but maybe there's a better workaround that I don't know of? Thanks!

**Comment by evesdropper** [comment]
Don't think there's a built-in solution, but I'm pretty sure you can try the following:
- Store the snippet you want as a return value in a function but leave the trigger key empty
- Create a table to store triggers you want for a specific snippet
- Iterate through the table and create a snippet.

For example, here is some skeleton code you can play around with:

```lua
local dummy_test = function(context, snippet, opts)
    opts = opts or {}
    if not context.trig then
        error("context doesn't include a `trig` key which is mandatory", 2)
    end 
    return s(context, snippet, opts)
end

local dummy_test_specs = {
    "dummy1", "dummy2", "dummy3" -- replace with other keywords you want to use
}
local dummy_snippet_body = {
    t("this is a dummy test snippet"),
    i(1), c(2, {t("more text"), i(1)})
}

local dummy_test_snippets = {}
for _, v in ipairs(dummy_test_specs) do
	table.insert(dummy_test_snippets, dummy_test({ trig = v }, dummy_snippet_body, { condition = in_text }))
end
vim.list_extend(M, dummy_test_snippets) -- this assumes M is the table of snippets; change if needed
```

And here is the code in action:
![dummytest](https://user-images.githubusercontent.com/82856360/219256729-1e4731e5-c3d4-4ae5-8bc1-0b6f30413517.gif)

**Reply by budimanjojo** [reply]
Thanks! I'll treat this as the most viable workaround for now.

**Comment by L3MON4D3** [comment]
Oh wow, great timing, I just worked on that xD
Check #774, it will support (amongst other things) multiple triggers.

**Reply by budimanjojo** [reply]
Awesome, thanks! Let us know when you're done :D
